espnet2/enh/decoder/seanet_decoder.py

```diff
@@ -0,0 +1,118 @@
+        encdec_activation: str = "ELU",
+        encdec_activation_params: Dict[str, Any] = {"alpha": 1.0},
+        encdec_norm: str = "weight_norm",
+        encdec_norm_params: Dict[str, Any] = {},
+        encdec_kernel_size: int = 7,
+        encdec_residual_kernel_size: int = 7,
+        encdec_last_kernel_size: int = 7,
+        encdec_dilation_base: int = 2,
+        encdec_causal: bool = False,
+        encdec_pad_mode: str = "reflect",
+        encdec_true_skip: bool = False,
+        encdec_compress: int = 2,
+        encdec_lstm: int = 2,
+        decoder_trim_right_ratio: float = 1.0,
+        decoder_final_activation: Optional[str] = None,
+        decoder_final_activation_params: Optional[dict] = None,
+    ):
+        super().__init__()
+        self.codec_decoder = SEANetDecoder(
+            channels=encdec_channels,
+            dimension=hidden_dim,
+            n_filters=encdec_n_filters,
+            n_residual_layers=encdec_n_residual_layers,
+            ratios=encdec_ratios,
+            activation=encdec_activation,
+            activation_params=encdec_activation_params,
+            norm=encdec_norm,
+            norm_params=encdec_norm_params,
+            kernel_size=encdec_kernel_size,
+            residual_kernel_size=encdec_residual_kernel_size,
+            last_kernel_size=encdec_last_kernel_size,
+            dilation_base=encdec_dilation_base,
+            causal=encdec_causal,
+            pad_mode=encdec_pad_mode,
+            true_skip=encdec_true_skip,
+            compress=encdec_compress,
+            lstm=encdec_lstm,
+            trim_right_ratio=decoder_trim_right_ratio,
+            final_activation=decoder_final_activation,
+            final_activation_params=decoder_final_activation_params,
+        )
+        self.codec_fs = codec_rate
+        self.sample_fs = sample_rate
+        if self.codec_fs != self.sample_fs:
+            self.dac_sampler = T.Resample(self.codec_fs, self.sample_fs)
+        self._output_dim = hidden_dim
+
+    def forward(self, input: torch.Tensor, ilens: torch.Tensor, fs: int = None):
+        """Forward.
+
+        Args:
+            input (torch.Tensor): spectrum [Batch, T, F]
+            ilens (torch.Tensor): input lengths [Batch] (not used)
+            fs (int): sampling rate in Hz (Not used)
+        """
+        wav = self.codec_decoder(input.transpose(1, 2))[:, 0, :]
+
+        # Resampling back to original sampling rate if needed
+        wav = wav.unsqueeze(1)
+        if self.codec_fs != self.sample_fs:
+            self.dac_sampler = self.dac_sampler.to(wav.device)
+            wav = self.resample_audio(wav)
+        wav = wav.squeeze(1)
+
+        # T might have changed due to model. If so, fix it here
+        T_origin = ilens.max()
+        if wav.shape[-1] != T_origin:
+
+            T_est = wav.shape[-1]
+            if T_origin > T_est:
+                wav = F.pad(wav, (0, T_origin - T_est))
+            else:
+                wav = wav[:, :T_origin]
```

Comment:
You can simply replace these lines with
```python
wav = F.pad(wav, (0, T_origin - wav.shape[-1]))
```

`F.pad` can automatically do padding or truncation according to the specified padding value.